File: LinkedList/PrintLinkedList.py
#Print elements of linked list


class Node:
    def __init__ (self,data,nextnode):
        self.data= data
        self.nextnode = nextnode

class LinkedList:
    def __init__ (self):
        self.currnode = None
    
    def addNode(self, data):
        node = Node(data,self.currnode)
        self.currnode = node
    
    def _print(self):
        node = self.currnode
        while node:
            print node.data
            node = node.nextnode

lk = LinkedList()
lk.addNode(3)
lk.addNode(4)
lk.addNode(5)
lk._print()

File: DS/LinkedList/1/PrintLinkedList.js
function Node(data, nextnode) {
    this.data = data;
    this.nextnode = nextnode;
}

function LinkedList() {
    this.currnode = null;
}

LinkedList.prototype = {
    addNode: function(data) {
        var newNode = new Node(data, this.currnode);
        this.currnode = newNode;
    },
    printList: function() {
        node = this.currnode;
        while (node) {
            console.log(node.data);
            node = node.nextnode;
        }
    }
}


var li = new LinkedList();
li.addNode(3);
li.addNode(5);
li.printList();